elks/arch/i86/kernel/irq-8259.c

```diff
@@ -18,17 +18,13 @@
 #define PIC2_CMD   0xA0   /* slave */
 #define PIC2_DATA  0xA1
 
-static unsigned char cache_21 = 0xff, cache_A1 = 0xff;
 
 /*
  *	Low level interrupt handling for the X86 PC/XT and PC/AT platform
  */
 
 void init_irq(void)
 {
-#ifdef CONFIG_HW_259_USE_ORIGINAL_MASK	/* for example Debugger :-) */
```

Comment:
Cleanup that configuration option.

Reply:
I am still a little worried about not using cached values here. In previous kernel, ELKS was the master and completely controlled system. In new approach, this leaves open for startup BIOS to have setup interrupts in certain way.

For embedded systems, this is probably a good idea, since the startup code is outside ELKS. 

For legacy IBM PC systems with pre v1 BIOS, etc. I don't know whether this is good idea.

Reply:
Well... maybe we could use EMU86 to monitor the PIC access from MS-DOS, and check it does not try to modify the interrupt masking ? If that monitoring shows that MS-DOS, the most widely deployed OS on the IBM PC, does not do that, maybe you could be convinced that ELKS has no reason to do it ?

Reply:
Interesting idea, but not certain it's really relevant to the time I have to spend maintaining ELKS on super old PCs that anybody tries to run ELKS on. 

I would like to know answer, can breakpoints be set on I/O ports in EMU86?

I am willing to live with caching removed in this PR, but leaving in enable_irq's on BIOS driver and others (except keyboard IRQ 1) with if 0 wrappers. Then at least it won't take too much of my time if you are wrong!

Reply:
> I would like to know answer, can breakpoints be set on I/O ports in EMU86?

No breakpoint as the existing ones on the command line, but it is easy to add a test in `io-elks.c` when accessing PIC I/O ports and break at that moment. Let me check MSDOS that way...

I will also surround all the removed pieces of code with 0 wrappers.

Reply:
Result from MSDOS monitoring : it installs some handlers for some traps (INT1, INT3 and INT4), sends one spurious EOI at boot, but **never** mask / unmask any interrupts...

Reply:
Well, that's possibly good news. MSDOS has a way of having its drivers spewn all over the place, in loadable TSRs, himem.sys, etc, so hard to tell for all devices. But I'm OK with leaving enable_irq's out but 0 wrapper, and we'll see. It is interesting to think that perhaps all the enable_irq previous code was all there just to keep up with ELKS own unneeded cached value of the PIC mask, in other words, bad design. And cached value, if not needed, isn't critical path since only read and written on request_irq or disable.

Reply:
I tried to use HIMEM.SYS and CD.SYS, but these modules require 80286 / 386 instructions and protected mode. At least it should bring confidence about the timer, keyboard, serial, parallel & floppy IRQs, as these devices are used though the BIOS during MSDOS boot.

Reply:
I am gaining confidence, but we will see.

At least with 0 wrapper approach, they could be easily changed to \#if CONFIG_BIOS_NEEDS_IRQ_ENABLE or something should this all be a problem. If not, then lots more excess cruft removed!

Reply:
Previous code wrapped in 5adc931